src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -607,7 +607,7 @@ namespace winrt::TerminalApp::implementation
                 {
                     for (auto i = tab->_mruPanes.begin(); i != tab->_mruPanes.end(); ++i)
                     {
-                        if (*i == pane->Id())
+                        if (pane->Id().has_value() && *i == pane->Id().value())

```

Comment:
optional has a cool feature: you can do a comparison directly with the optional.

```suggestion
                        if (*i == pane->Id())
```

except here's the difference: it will be _false_ if `Id` is nullopt.